frontend/lib/resolution/Resolver.cpp

```diff
@@ -2280,11 +2286,15 @@ QualifiedType Resolver::typeForEnumElement(const EnumType* enumType,
                                  /* receiverScope */ nullptr,
                                  elementName, config);
     if (vec.size() == 0) {
-      return typeErr(nodeForErr, "no enum element with given name");
+      return CHPL_TYPE_ERROR(context, UnknownEnumElem, nodeForErr,
+                             elementName, enumType);
     } else if (vec.size() > 1 || vec[0].numIds() > 1) {
+      auto& ids = vec[0];
       // multiple candidates. report a type error, but the
       // expression most likely has a type given by the enum.
-      typeErr(nodeForErr, "duplicate enum elements with given name");
+      std::vector<ID> redefinedIds(ids.numIds());
+      std::copy(ids.begin(), ids.end(), redefinedIds.begin());
```

Comment:
What's the reason for manually copying and then moving this, as opposed to just passing the vector in normally? Since the vector and its elements are all on the stack won't it just copy anyways? (Moreover, could it just be passed as a const ref?)

Reply:
Well, it's not quite a vector, but a `BorrowedIdsWithName`. I could pass _that_ in, I suppose, but my gut feeling was to avoid "references that could expire" (which is the case with BorrowedIds, since it refers to a raw vector pointer that belongs to something else). I can't use a const ref because the error needs something with a longer lifetime (`vec` is allocated on the stack, errors are reported only after resolving is done).